File: lib/dartFiles/Assignment.dart
import 'Course.dart';

class Assignment {
  String name;
  String description;
  double score;
  Course course;
  DateTime deadline;
  int remainingDays = 0;
  bool isActive = false;

  Assignment(this.course, String deadline, this.name, this.description, this.score)
      : deadline = DateTime.parse('yyyy-MM-dd');

  void setDeadline(String deadline) {
    this.deadline = DateTime.parse('yyyy-MM-dd');
  }

  DateTime get getDeadline => deadline;

  void _updateRemainingDays() {
    final currentDate = DateTime.now();
    remainingDays = deadline.difference(currentDate).inDays;
  }

  void printRemainingDays() {
    if (!isActive) {
      print('This assignment is not active yet.');
      return;
    }
    _updateRemainingDays();
    if (remainingDays > 0) {
      print('You have $remainingDays days to do the assignment.');
    } else if (remainingDays == 0) {
      print('Deadline is today !!');
    } else {
      print('Deadline has already ended :(');
    }
  }

  void setActive(bool isActive) {
    this.isActive = isActive;
  }

  bool get getIsActive => isActive;

  Course get getCourse => course;

  String get getDescription => description;

  String get getName => name;

  double get getScore => score;

  int get getRemainingDays {
    _updateRemainingDays();
    return remainingDays;
  }
}

File: Backend/src/miniP/Assignment.java
package miniP;

import java.io.Serializable;
import java.time.Duration;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class Assignment implements Serializable {
    private String name;
    private String description;
    private double score;
    private Course course;
    private LocalDateTime deadline;
    private long remainingDays;
    private boolean isActive;

    public Assignment(Course course, String deadline, String name, String description, double score){
        this.course = course;
        this.deadline = LocalDateTime.parse(deadline);
        this.name = name;
        this.description = description;
        this.score = score;
    }

    public String serializeNotFinished() {
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS");
        return name + "," + description + "," +  deadline.format(formatter) + "," + "false," + score + '\n';
    }

    public String serializeAll() {
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS");
        return name + "," + description + "," +  deadline.format(formatter) + "," + "false," + score + '\n';
    }

    public void setDeadline(String deadline) {
        this.deadline = LocalDateTime.parse(deadline);
    }

    public LocalDateTime getDeadline() {
        return deadline;
    }

    private void updateRemainingDays() {
        LocalDateTime currentDate = LocalDateTime.now();
        remainingDays = Duration.between(currentDate, deadline).toDays();
    }

    public void printRemainingDays() {
        if (!isActive()){
            System.out.println("This assignment is not active yet.");
            return;
        }
        updateRemainingDays();
        if (remainingDays > 0)
            System.out.println("You have " + remainingDays + " days to do the assignment.");
        else if (remainingDays == 0)
            System.out.println("Deadline is today !!");
        else
            System.out.println("deadline has already ended :(");
    }

    public void setActive(boolean isActive) {
        this.isActive = isActive;
    }

    public boolean isActive() {
        return isActive;
    }

    public Course getCourse() {
        return course;
    }

    public String getDescription() {
        return description;
    }

    public String getName() {
        return name;
    }

    public double getScore() {
        return score;
    }

    public long getRemainingDays() {
        updateRemainingDays();
        return remainingDays;
    }
}
